File: 2586-count-the-number-of-vowel-strings-in-range/2586-count-the-number-of-vowel-strings-in-range.py
class Solution:
    def isVowel(self, ch) :
        if ch == 'a' or ch == 'e' or ch == 'i' or ch == 'o' or ch == 'u' :
            return True
        return False 

    def vowelStrings(self, words: list[str], left: int, right: int) -> int :
        '''
            Time complexity = O(right - left + 1) = O(k)
            Space complexity = O(1)
        '''

        n = len(words)
        count_vowel_strings = 0

        for i in range(left, right + 1) :
            word = words[i]
            m = len(word)

            left_ch = word[0]
            right_ch = word[m - 1]

            if(self.isVowel(left_ch) and self.isVowel(right_ch)) :
                count_vowel_strings += 1

        return count_vowel_strings

File: 2586-count-the-number-of-vowel-strings-in-range/2586-count-the-number-of-vowel-strings-in-range.java
class Solution {
    public boolean isVowel(char ch){
        if(ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u') 
        {
            return true;
        }

        return false;
    }

    public int vowelStrings(String[] words, int left, int right) {
        /*
            Time complexity = O(right - left + 1) = O(k), 
            Space complexity = O(1)
        */
        
        int n = words.length;
        int countVowelStrings = 0;

        for(int i = left; i <= right; i++){
            String word = words[i];
            int m = word.length();

            char leftCh = word.charAt(0);
            char rightCh = word.charAt(m - 1);

            if(isVowel(leftCh) && isVowel(rightCh)){
                countVowelStrings++;
            }
        }

        return countVowelStrings;
    }
}